File: beaglesightlibs/src/testDebug/aidl/com/cross/beaglesightlibs/LineOfBestFitCalculatorTest.java
package com.cross.beaglesightlibs;

import com.cross.beaglesightlibs.exceptions.InvalidNumberFormatException;

import junit.framework.Assert;

import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;


class LineOfBestFitCalculatorTest {

    @Test
    void calcPosition() throws InvalidNumberFormatException {
        {
            List<PositionPair> pos = new ArrayList<>();
            pos.add(new PositionPair("10", "10"));
            LineOfBestFitCalculator calc = new LineOfBestFitCalculator();
            calc.setPositions(pos);
            Assert.assertEquals(Double.NaN, calc.calcPosition(11));
        }
        {
            List<PositionPair> pos = new ArrayList<>();
            pos.add(new PositionPair("10", "10"));
            pos.add(new PositionPair("20", "20"));
            LineOfBestFitCalculator calc = new LineOfBestFitCalculator();
            calc.setPositions(pos);
            Assert.assertEquals(15.0, calc.calcPosition(15));
        }
    }
}

File: app/beaglesightlibs/src/test/java/com/cross/beaglesightlibs/LineOfBestFitCalculatorTest.kt
package com.cross.beaglesightlibs

import com.cross.beaglesightlibs.bowconfigs.LineOfBestFitCalculator
import com.cross.beaglesightlibs.bowconfigs.PositionPair
import com.cross.beaglesightlibs.exceptions.InvalidNumberFormatException
import org.junit.Assert
import org.junit.Test

class LineOfBestFitCalculatorTest {
    @Test
    @Throws(InvalidNumberFormatException::class)
    fun calcPosition() {
        run {
            val pos: MutableList<PositionPair> = ArrayList()
            pos.add(PositionPair(10.0f, 10.0f))
            val calc = LineOfBestFitCalculator()
            calc.positions = pos
            Assert.assertEquals(Double.NaN, calc.calcPosition(11.0f).toDouble(), 0.001)
        }
        run {
            val pos: MutableList<PositionPair> = ArrayList()
            pos.add(PositionPair(10.0f, 10.0f))
            pos.add(PositionPair(20.0f, 20.0f))
            val calc = LineOfBestFitCalculator()
            calc.positions = pos
            Assert.assertEquals(15.0f, calc.calcPosition(15.0f), 0.001f)
        }
    }
}